packages/coffeescript/plugin/compile-coffeescript.js

```diff
@@ -83,8 +83,9 @@ export class CoffeeCompiler extends CachingCompiler {
 
     let sourceMap = JSON.parse(output.v3SourceMap);
 
-    if (source.indexOf('`') !== -1) {
-      // If source contains backticks, pass the coffee output through babel-compiler
+    if (source.match(/(`|import|export|function\*)/)) {
```

Comment:
If I'm not mistaken, `source` is the contents of a CoffeeScript file (_not_ JavaScript) and CoffeeScript doesn't have the `function*` generator syntax. [Per CoffeeScript docs, they refer to it as "nonsense"](http://coffeescript.org/#fat-arrow):

> CoffeeScript functions also support ES6 generator functions through the `yield` keyword. There's no `function*(){}` nonsense — a generator in CoffeeScript is simply a function that yields.

Therefore, this should be looking for `yield` instead of `function*`, no?